optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/ConstraintSession.java

```diff
@@ -26,7 +26,7 @@
 /**
  * An instance of this class must be used in only one thread.
  */
-public interface ConstraintSession<Solution_>  {
+public interface ConstraintSession<Solution_> {
```

Comment:
Shouldn't this rather implement `Closeable` or `AutoCloseable`? That would result in the IDE giving useful tips regarding forgetting to close it.

Reply:
agreed